### ISSUE TITLE
[Phase 1] Data Loading & Validation

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/real-estate-valuation-engine#3]
## Summary

Implement data loading from CSV files with schema validation and coordinate parsing into GeoDataFrame format.

**Part of:** #1
**Depends on:** #2 (Phase 0)

## Objectives

- [ ] Implement `DataLoader.load_csv()` method
- [ ] Create Pydantic validation schemas
- [ ] Handle coordinate parsing to geometry
- [ ] Add custom exceptions
- [ ] Write unit tests for happy path and error cases

## Files to Modify

| File | Lines | Action | Description |
|------|-------|--------|-------------|
| `src/data/loader.py` | 1-80 | Implement | Complete DataLoader class |
| `src/data/schema.py` | 1-40 | Review | PropertyRecord validation |
| `src/data/exceptions.py` | 1-20 | Review | Custom exceptions |
| `tests/data/test_loader.py` | NEW | Create | Unit tests |

### DataLoader.load_csv()

```python
def load_csv(self, filepath: Union[str, Path]) -> gpd.GeoDataFrame:
    """Load property data from CSV file."""
    filepath = Path(filepath)

    # Check file exists
    if not filepath.exists():
        raise DataLoadError(f"File not found: {filepath}")

    # Load CSV with pandas
    try:
        self._raw_df = pd.read_csv(filepath, parse_dates=['sale_date'])
    except Exception as e:
        raise DataLoadError(f"Failed to parse CSV: {e}")

    # Validate records if enabled
    if self.validate:
        self._validate_records()

    # Convert to GeoDataFrame with geometry column
    self._gdf = gpd.GeoDataFrame(
        self._raw_df,
        geometry=gpd.points_from_xy(
            self._raw_df.longitude,
            self._raw_df.latitude
        ),
        crs="EPSG:4326"  # WGS84
    )

    return self._gdf
```

### Test: Reject Invalid Coordinates

```python
# tests/data/test_loader.py
import pytest
from src.data.loader import DataLoader
from src.data.exceptions import SchemaValidationError

def test_loader_rejects_invalid_coordinates(tmp_path):
    """DataLoader should reject records with out-of-range coordinates."""
    # Create invalid CSV
    csv_content = """property_id,latitude,longitude,sale_price,sale_date
P001,999.0,-122.4,500000,2023-01-15"""

    csv_file = tmp_path / "invalid.csv"
    csv_file.write_text(csv_content)

    loader = DataLoader(validate=True)

    with pytest.raises(SchemaValidationError):
        loader.load_csv(csv_file)
```

## Verification

```bash
# Run specific tests
pytest tests/data/test_loader.py -v

# Test loading sample data
python -c "
from src.data.loader import DataLoader
loader = DataLoader(validate=True)
gdf = loader.load_csv('tests/fixtures/sample_properties.csv')
print(f'Loaded {len(gdf)} records')
print(f'CRS: {gdf.crs}')
print(loader.summary)
"
```

## Technical Challenges

1. **Date parsing edge cases** - Different date formats
   - Solution: Use `pd.to_datetime` with `errors='coerce'`
2. **Coordinate system confusion** - Lat/long order varies
   - Solution: Explicit column naming, validate ranges

## Definition of Done

- [ ] Sample CSV loads successfully
- [ ] Invalid coordinates raise SchemaValidationError
- [ ] GeoDataFrame has valid geometry column with CRS EPSG:4326
- [ ] All unit tests pass
- [ ] Summary property returns expected statistics

## Labels

`phase-1`, `data`, `enhancement`
